src/lib/wordpress-provider/index.ts

```diff
@@ -0,0 +1,83 @@
+export * from './types';
+export { WpNowProvider } from './wp-now';
+
+import { WpNowProvider } from './wp-now';
+import type { WordPressProvider } from './types';
+
+let provider: WordPressProvider | null = null;
+let providerType: string = 'wp-now'; // Default provider type
+
+export function getWordPressProvider(): WordPressProvider {
+	if ( ! provider ) {
+		provider = new WpNowProvider();
+	}
+	return provider;
+}
+
+export function getWordPressProviderType(): string {
+	return providerType;
+}
+
+export function setWordPressProvider( newProvider: WordPressProvider, type?: string ): void {
+	provider = newProvider;
+	if ( type ) {
+		providerType = type;
+	}
+
+	// Notify renderer processes about provider constants changes
+	try {
+		const { sendIpcEventToRenderer } = require( 'src/ipc-utils' );
+		const constants = {
+			defaultPhpVersion: DEFAULT_PHP_VERSION(),
+			defaultWordPressVersion: DEFAULT_WORDPRESS_VERSION(),
+			allowedPhpVersions: ALLOWED_PHP_VERSIONS(),
+		};
```

Comment:
Using the provider can simplify the logic:

```suggestion
		const constants = {
			defaultPhpVersion: provider.DEFAULT_PHP_VERSION,
			defaultWordPressVersion: provider.DEFAULT_WORDPRESS_VERSION,
			allowedPhpVersions: provider.ALLOWED_PHP_VERSIONS,
		};
```